tests/app/questionnaire/test_path_finder.py

```diff
@@ -1,720 +1,632 @@
+    )
+
+    expected_path = RoutingPath(
+        ["proxy", "date-of-birth", "confirm-dob", "sex"],
+        section_id="personal-details-section",
+        list_name="people",
+        list_item_id="abc123",
+    )
+
+    assert routing_path == expected_path
+
+
+def test_routing_path_empty_routing_rules(answer_store, list_store):
+    schema = load_schema_from_name("test_checkbox")
+    section_id = schema.get_section_id_for_block_id("mandatory-checkbox")
+    expected_path = RoutingPath(
+        ["mandatory-checkbox", "non-mandatory-checkbox", "single-checkbox"],
+        section_id="default-section",
+    )
+
+    answer_1 = Answer(answer_id="mandatory-checkbox-answer", value="Cheese")
+    answer_2 = Answer(answer_id="non-mandatory-checkbox-answer", value="deep pan")
+    answer_3 = Answer(answer_id="single-checkbox-answer", value="Estimate")
+
+    answer_store.add_or_update(answer_1)
+    answer_store.add_or_update(answer_2)
+    answer_store.add_or_update(answer_3)
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "default-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["mandatory-checkbox"],
+            }
+        ]
+    )
+
+    path_finder = PathFinder(schema, answer_store, list_store, progress_store, {}, {})
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    assert routing_path == expected_path
+
+
+def test_routing_path_with_conditional_value_not_in_metadata(answer_store, list_store):
+    schema = load_schema_from_name("test_metadata_routing")
+    section_id = schema.get_section_id_for_block_id("block1")
+    expected_path = RoutingPath(
+        ["block1", "block2", "block3"], section_id="default-section"
+    )
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "default-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["block1"],
+            }
+        ]
+    )
+
+    path_finder = PathFinder(schema, answer_store, list_store, progress_store, {}, {})
+    routing_path = path_finder.routing_path(section_id=section_id)
+
+    assert routing_path == expected_path
+
+
+def test_routing_path_should_skip_block(answer_store, list_store):
+    # Given
+    schema = load_schema_from_name("test_skip_condition_block")
+    section_id = schema.get_section_id_for_block_id("should-skip")
+    answer_store.add_or_update(
+        Answer(answer_id="do-you-want-to-skip-answer", value="Yes")
+    )
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "introduction-section",
+                "list_item_id": None,
+                "status": CompletionStatus.COMPLETED,
+                "block_ids": ["do-you-want-to-skip"],
+            }
+        ]
+    )
+
+    # When
+    path_finder = PathFinder(
```

Comment:
All these path_finders look the same, we should try to extract it to conftest and pass answer_store and progress_store in

Reply:
They do, but half of the params are used, by the time I've made a fixture from this, it ends up being just another constructor for `PathFinder`, that's why I avoided it in favour of not obscuring what's being created.

Reply:
No worries